File: services/+logging/agent_log_writer.js
// @service: agent_log_writer
// @role: Appends structured log lines to agent_calls log file

const fs = require('fs');
const path = require('path');

const LOG_DIR = "logs/agent_calls";

// Ensure log directory exists
if (!fs.existsSync(LOG_DIR)) {
  fs.mkdirSync(LOG_DIR, { recursive: true });
}

function logAgentEvent(message) {
  const now = new Date();
  const timestamp = now.toISOString().split('T')[0]; // YYYY-MM-DD
  const time = now.toTimeString().split(' ')[0].substring(0, 5); // HH:MM
  const logFile = path.join(LOG_DIR, `${timestamp}_agent_calls.log`);

  const entry = `[${time}] ${message}\n`;
  fs.appendFileSync(logFile, entry, 'utf8');
}

// Example usage:
// logAgentEvent("ClaudeCode called: Hera → panels/A1_GameTipsPanel");

module.exports = { logAgentEvent };

File: services/+logging/agent_log_writer.py
# @service: agent_log_writer
# @role: Appends structured log lines to agent_calls log file

import os
from datetime import datetime

LOG_DIR = "logs/agent_calls"
os.makedirs(LOG_DIR, exist_ok=True)

def log_agent_event(message: str):
    """Log an agent event with timestamp to daily log file"""
    timestamp = datetime.now().strftime("%Y-%m-%d")
    time = datetime.now().strftime("%H:%M")
    log_file = os.path.join(LOG_DIR, f"{timestamp}_agent_calls.log")

    entry = f"[{time}] {message}\n"
    with open(log_file, "a") as f:
        f.write(entry)

# Example usage:
# log_agent_event("ClaudeCode called: Hera → panels/A1_GameTipsPanel")